smac/intensification/intensification.py

```diff
@@ -171,60 +502,45 @@ def eval_challenger(self,
+                )
                 incumbent = challenger
-            else:
-                inc_runs = run_history.get_runs_for_config(incumbent, only_max_observed_budget=True)
-                if len(inc_runs) > 0:
-                    self.logger.debug("Skipping RUN_FIRST_CONFIG stage since incumbent is already run before")
-                    self.stage = IntensifierStage.RUN_INCUMBENT
 
-        self.logger.debug("Intensify on %s", challenger)
-        if hasattr(challenger, 'origin'):
-            self.logger.debug("Configuration origin: %s", challenger.origin)
+        if self.stage in [IntensifierStage.PROCESS_INCUMBENT_RUN,
+                          IntensifierStage.PROCESS_FIRST_CONFIG_RUN]:
+            self._ta_time += result.time
+            self.num_run += 1
+            self._process_inc_run(
+                incumbent=incumbent,
+                run_history=run_history,
+                log_traj=log_traj,
+            )
 
-        try:
-            # since it runs only 1 "ExecuteRun" per iteration,
-            # we run incumbent once and then challenger in the next
-            if self.stage == IntensifierStage.RUN_FIRST_CONFIG or \
-                    self.stage == IntensifierStage.RUN_INCUMBENT:
-                # Lines 3-7
-                self._add_inc_run(incumbent=incumbent, run_history=run_history, log_traj=log_traj)
-            elif self.stage == IntensifierStage.RUN_CHALLENGER or \
-                    self.stage == IntensifierStage.RUN_BASIS:
-                # Lines 8-17
-                incumbent = self._race_challenger(challenger=challenger,
-                                                  incumbent=incumbent,
-                                                  run_history=run_history,
-                                                  log_traj=log_traj)
-
-        except BudgetExhaustedException:
-            # We return incumbent, SMBO stops due to its own budget checks
-            inc_perf = run_history.get_cost(incumbent)
-            self.logger.debug("Budget exhausted; Return incumbent")
-            return incumbent, inc_perf
-
-        # time used to evaluate challenger
-        self.elapsed_time += time.time() - start_time
+        else:
+            self.num_run += 1
+            self.num_chall_run += 1
+            if result.status == StatusType.CAPPED:
+                # move on to the next iteration
+                self.logger.debug(
+                    "Challenger itensification timed out due "
+                    "to adaptive capping."
+                )
+                self.stage = IntensifierStage.RUN_INCUMBENT
+            else:
+
+                self._ta_time += result.time
+                incumbent = self._process_racer_results(
+                    challenger=challenger,
+                    incumbent=incumbent,
+                    run_history=run_history,
+                    log_traj=log_traj,
+                )
 
+        elapsed_time = result.endtime - result.starttime
```

Comment:
I am wondering two things here: 
1) What happened to self.elapsed time? It is still an attribute of this class but is is never updated, however, I don't see where we use it?
2) What is the difference between result.endtime - result.starttime and result.time

Reply:
Thanks, this clearly has to be cleaned. Let me do a push for this.

In the old code, there are 2 mechanisms to see if there is time left for this iteration:

- Trusting the evaluator implemented by the user (use_ta_time_bound is set to true), so the time this iteration has gone is constructed by adding the duration stated in the runhistory time attribute
- Measuring the duration via time.time() at the start (result.starttime) and at the end (result.endtime ) regardless of what the ta function does, and use this to measure the accumulative duration of this iteration.

So to answer your point 2, the difference is whether you trust the ta run method implemented by the user of smac, or you just track it yourself using the time module.


Elapsed time, in the old code measure the time it took to evaluate the challenger configuration